x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -231,43 +236,44 @@ function createGetTransactionById(itemsById: IWaterfallIndex) {
 }
 
 export function getWaterfall(
-  hits: Array<Span | Transaction>,
-  entryTransaction: Transaction
+  hits: TraceAPIResponse['trace'],
+  errorsPerTransaction: TraceAPIResponse['errorsPerTransaction'],
+  entryTransactionId?: Transaction['transaction']['id']
 ): IWaterfall {
-  if (isEmpty(hits)) {
+  if (isEmpty(hits) || !entryTransactionId) {
     return {
       services: [],
       duration: 0,
       items: [],
       itemsById: {},
       getTransactionById: () => undefined,
+      errorsPerTransaction,
       serviceColors: {}
     };
```

Comment:
Just a thought: would it make sense to `return undefined` here and exit early in the view code?